src/Http/Http/src/DefaultHttpContextFactory.cs

```diff
@@ -0,0 +1,64 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.AspNetCore.Http.Features;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.AspNetCore.Http
+{
+    public class DefaultHttpContextFactory : IHttpContextFactory
```

Comment:
Do we really need this to be public? Should this be sealed or is this documented as extensible?

Reply:
Don't know about being public, but sealing a class that has no virtual methods has no benefit?

Reply:
The benefit is not having to reason about whether or not other inherited from it when we make changes in the future. If there's no actual extensibility points (no virtual methods) then why not give ourselves the most flexibility to make changes in the future?

Basically @davidfowl has been razzing me about my designs for like 2 years. I gotta score points on him where I can.

Reply:
That's fair, it could be made internal but it feels a little gross since users can new it up and use it in their own factory. Though that's gross as well.